src/main/scala/beam/router/model/EmbodiedBeamTrip.scala

```diff
@@ -115,7 +118,10 @@ object EmbodiedBeamTrip {
         theMode = leg.beamLeg.mode
       } else if (theMode == WALK && leg.beamLeg.mode == BIKE) {
         theMode = leg.beamLeg.mode
+      } else if (theMode == WALK && leg.beamLeg.mode == EMERGENCY) {
+        theMode = EMERGENCY
```

Comment:
1. Because we introduce a new possible return value for `EmbodiedBeamTrip#tripClassifier` then we need to carefully check all the pattern matching on this field to validate that there wouldn't be a `MatchError`.
2. Also because EMERGENCY trip is actually a WALK trip. Then we need to check all the places where `tripClassifer` is compared with WALK (i.e. https://github.com/LBNL-UCB-STI/beam/blob/676c68782f48780c0f00ad797b6fa9c56bf6f018/src/main/scala/beam/agentsim/agents/PersonAgent.scala#L264) and add these EMERGENCY there.